src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -677,6 +677,74 @@ class npc_greengill_slave : public CreatureScript
     };
 };
 
+/*######
+## npc_irespeaker
+######*/
+
+enum Spells
+{
+    SPELL_FEL_FIREBALL = 35913,
+    SPELL_CURSE_OF_WEAKNESS = 18267,
+    SPELL_FEL_CONSUMPTION = 45023,
+};
+
+struct npc_irespeaker : public ScriptedAI
+{
+    npc_irespeaker(Creature* creature) : ScriptedAI(creature), _scheduler() {  }
+
+    void JustEngagedWith(Unit* /*who*/ ) override
+    {
+        ScheduleFelFireball(); // Start with Fel Fireball if initially out of melee range
+        ScheduleCurseOfWeakness(); // Start with Curse of Weakness if in melee range
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        me->CastSpell(me, SPELL_FEL_CONSUMPTION, true);
+        _scheduler.CancelAll(); // Cancel all tasks on death
```

Comment:
Tasks should be reset either on Reset() or JustEngaged() (before they're scheduled)
Timers won't update while the creature is dead